urlcdab7d8364_jrwalsh_CycTools_tgz-pJ8-edu_iastate_cyctools_MainJ8/src/edu/iastate/cyctools/tools/load/model/DocumentModel.java

```diff
@@ -0,0 +1,102 @@
+package edu.iastate.cyctools.tools.load.model;
+
+import edu.iastate.cyctools.DefaultController;
+import edu.iastate.cyctools.externalSourceCode.AbstractModel;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+
+import javax.swing.table.DefaultTableModel;
+import javax.swing.table.TableModel;
+
+public class DocumentModel extends AbstractModel {
+    private File file;
+    private TableModel tableModel;
+    private String delimiter;
+    
+    public DocumentModel() {
+    }
+    
+    public void initDefault() {
+        setFile(null, ",");
+        setTableModel(null);
+    }
+    
+    // Accessors
+    public File getFile() {
+        return file;
+    }
+    
+    public void setFile(File file, String delimiter) {
+        File oldFile = this.file;
+        this.file = file;
+        this.delimiter = delimiter;
+        
+        readFile();
+        
+        firePropertyChange(DefaultController.DOCUMENT_FILEPATH_PROPERTY, oldFile, file);
+    }
+
+    public TableModel getTableModel() {
+        return tableModel;
+    }
+
+    private void setTableModel(TableModel tableModel) {
+        TableModel oldTableModel = this.tableModel;
+        this.tableModel = tableModel;
+        firePropertyChange(DefaultController.DOCUMENT_TABLEMODEL_PROPERTY, oldTableModel, tableModel);
+    }
+
+    // Utilities
+    private void readFile() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.